src/main/java/com/ssoyeong/studyapplication/controller/TestController.java

```diff
@@ -1,19 +1,26 @@
 package com.ssoyeong.studyapplication.controller;
 
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 
 @RestController
 public class TestController {
+    private static String sayHello = "Hello, %s!";
 
     @GetMapping("/test/hello")
     public String getHello() {
         return "Hello world!";
     }
 
     @GetMapping("/test/greeting")
-    public String getGreeting(@RequestParam String name) {
-        return String.format("Hello, %s!", name);
+    public String getGreetingQueryString(@RequestParam String name) {
+        return String.format(sayHello, name);
```

Comment:
```suggestion
        return sayHello.formatted(name);
```

Java 15에서 추가 된 문자열 포메팅 메서드입니다.
`formatted()`가 어떻게 구현되어 있고, String.format()과 어떻게 다른지 가볍게 보고 넘어가면 좋을 것 같습니다.

Reply:
`String.format`과 `formatted()`에 대해 알아보았는데, 제가 작성한 코드에서 `formatted()` 사용을 제안하신 이유를 알고 싶습니다.

`String.format`은 static 메서드이고 `formatted()`는 instance 메서드라고 조사했습니다.
"Hello, %s!"라는 문자열은 공통된 값을 유지하면서 TestController 클래스에서 여러 번 사용되니까 static 메서드인 `String.format`을 사용하는 게 바람직(?)하지 않을까?! 라고 생각했습니다.

개념적인 차이는 알겠는데 실제 코드에 왜 이렇게 적용해야하나..라는 고민이 들었습니다.

Reply:
Java 15에서 추가된 String.formatted() 메서드는 문자열 형식화를 간편하게 수행하는 데 도움이 되는 메서드입니다. 이 메서드를 사용하는 것에는 몇 가지 이점이 있습니다:

가독성 향상: String.format()을 사용하는 대신 String.formatted()를 사용하면 코드의 가독성이 향상됩니다. formatted() 메서드는 가독성이 높은 메서드 이름을 갖추고 있어서 형식화된 문자열이 어떻게 보일지 더 명확하게 나타낼 수 있습니다.

```java
// 이전 방식
String result = String.format("Hello, %s! Today is %s.", name, day);

// 새로운 방식 (Java 15+)
String result = "Hello, %s! Today is %s.".formatted(name, day);
```

불변성 유지: String.formatted()는 새로운 문자열을 생성하여 반환하므로 원래 문자열의 불변성을 유지합니다. 기존의 String.format()은 StringBuilder를 사용하여 문자열을 만들었기 때문에 새로운 문자열을 생성할 때마다 객체를 생성하고 버리는 오버헤드가 있을 수 있습니다.

```java
String result = "Hello, %s! Today is %s.".formatted(name, day);
```

Null 안전성: String.formatted() 메서드는 null 값을 다룰 때 더 안전하게 작동합니다. 형식화할 인수 중 하나가 null이면 %s로 처리되어 null 대신 문자열 "null"을 결과로 반환합니다.

```java
String result = "Hello, %s! Today is %s.".formatted(null, day);
// 결과: "Hello, null! Today is dayValue."
```

Localizable 문자열 처리: 이 메서드는 Locale을 사용하여 지역화된 문자열을 작성할 수 있는 기능을 제공합니다.

```java
String result = "Total: %,.2f".formatted(12345.6789);
// 결과 (예시): "Total: 12,345.68"
```

이러한 이점들은 코드 작성과 유지보수를 더 쉽게 만들고, 가독성을 향상시키며, 불변성을 유지하면서 현대적인 코드를 작성하는 데 도움이 됩니다.